sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+object RegExpShim {
+  // Handle regexp_replace inconsistency from https://issues.apache.org/jira/browse/SPARK-39107
+  def reproduceEmptyStringBug(): Boolean = false
```

Comment:
This is a fourth occurrence of this shim object with the same impl: 314, 322, 323, 330+. Can we create a new dir `post314` and include it in
in the minimum amount of sparkXYZ.pattern and sparkXYZ.sources.

Reply:
I plan to move this `reproduceEmptyStringBug()` to `SparkShimImpl`, and remove all the files named `RegExpShim.scala` in a following PR.

Reply:
`post314` apears to cover 320, 321, but they have a different impl.

Reply:
I used post314 to distinguish it from 314+. Similar how we with pre320-treenode and post320-treenode

Reply:
Follow up issue: https://github.com/NVIDIA/spark-rapids/issues/7105